MaterialDesignThemes.Wpf/PopupBox.cs

```diff
@@ -439,21 +453,67 @@ public override void OnApplyTemplate()
             VisualStateManager.GoToState(this, IsPopupOpen ? PopupIsOpenStateName : PopupIsClosedStateName, false);
         }
 
-        protected override void OnIsKeyboardFocusWithinChanged(DependencyPropertyChangedEventArgs e)
+        bool supressfirstElementGotFocus = true;
+        IInputElement firstchildelement = null;
+
+        protected override void OnGotKeyboardFocus(KeyboardFocusChangedEventArgs e)
         {
-            base.OnIsKeyboardFocusWithinChanged(e);
+            base.OnGotKeyboardFocus(e);
+            if (supressfirstElementGotFocus == false) { return; }
 
-            if (IsPopupOpen && !IsKeyboardFocusWithin && !StaysOpen)
+            if (!IsPopupOpen && (PopupMode == PopupBoxPopupMode.ClickOrFocusWithin | 
+                                 PopupMode == PopupBoxPopupMode.MouseOverEagerOrFocusWithin | PopupMode == PopupBoxPopupMode.MouseOverOrFocusWithin)
+                             && IsKeyboardFocusWithin)
             {
-                Close();
+                SetCurrentValue(IsPopupOpenProperty, true);
+                if (_popupContentControl != null)
+                {
+                    _popupContentControl.MoveFocus(new TraversalRequest(FocusNavigationDirection.First));
+
+                    //getting the first element, and ensure that when it gets focus again, move the focus outside of the Popup
+                    //to avoid unwanted tab stop cycling only inside it.
+                    firstchildelement = Keyboard.FocusedElement;
+                    if (firstchildelement == null) { return; }
+                    firstchildelement.GotKeyboardFocus += InnerFirstElement_GotKeyboardFocus;
+                    firstchildelement.LostKeyboardFocus += InnerFirstElement_LostKeyboardFocus;
```

Comment:
I would prefer
```
firstchildelement?.GotKeyboardFocus += InnerFirstElement_GotKeyboardFocus;
firstchildelement?.LostKeyboardFocus += InnerFirstElement_LostKeyboardFocus;
```
as the null checks simply serves to prevent exceptions.

Reply:
I'll try to resolve this. Otherwise I'll change to a common if (condition) {}

Reply:
see above reply

Reply:
Changes done. Thanks.